viewing_party/party.py

```diff
@@ -1,23 +1,180 @@
+    user_data = {"watched":[movie]}
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist": [movie]}
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+    else:
+            return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total_rating = 0
+    index = 0 
+    if user_data == {"watched": []}:
+        return 0.0
+    else:
+        for movie in user_data["watched"]:
+            index += 1
+            total_rating += (user_data["watched"][index-1]["rating"])
+        average_rating = total_rating/index
+        return average_rating
 
+def get_most_watched_genre(user_data):
+    index = 0
+    genres = []
+    least_popular = []
+    most_popular = []
+    if user_data == {"watched": []}:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            index += 1
+            genres.append(user_data["watched"][index-1]["genre"])
+        for item in genres:
+            if item not in least_popular:
+                least_popular.append(item)
+            else:
+                most_popular.append(item)
+        return (most_popular[0])
+            
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    unique = []
+    total_movies_list = []
+    users_movies = []
+    friends_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    total_movies_list = users_movies + friends_movies
+
+    for item in total_movies_list:
+        if item not in friends_movies:
+            unique.append(item)
+    return unique
+
+def get_friends_unique_watched(user_data):
+    friends_unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in friends_unique:
+                friends_unique.append(item)
+    return friends_unique
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended = []
+    users_movies = []
+    friends_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)

```

Comment:
Is it necessary to make this list? `users_movies` will have the same data as `user_data["watched"]`. A helper variable is a great idea here, but unless some other part of the algorithm is modifying `users_movies`, it would be sufficient to set the variable equal to the existing list:
```suggestion
    users_movies =  user_data['watched']
```